From 703c6115445acd7c136dc3fdffa0c700be4a7091 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Fri, 9 Feb 2018 01:05:43 +0700
Subject: [PATCH] Problem: recording files in current directory fails

If the current directory is outside of the working directory, sit will
not allow to have these recorded.

Solution: use current directory instead of working directory
as a way to detect where the files are.
---
 sit/src/main.rs | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sit/src/main.rs b/sit/src/main.rs
index 1e50d15..93c38e9 100644
--- a/sit/src/main.rs
+++ b/sit/src/main.rs
@@ -366,9 +366,10 @@ fn main() {
                                     exit(1);
                                 }
                                 let abs_name = fs::canonicalize(path).expect("can't canonicalize path");
-                                match abs_name.strip_prefix(&canonical_working_dir) {
+                                let cur_dir = fs::canonicalize(env::current_dir().expect("can't get current directory")).expect("can't canonicalize current directory");
+                                match abs_name.strip_prefix(&cur_dir) {
                                     Err(_) => {
-                                        eprintln!("Path {} is not relative to {} working directory", name, canonical_working_dir.to_str().unwrap());
+                                        eprintln!("Path {} is not relative to {} current directory", name, cur_dir.to_str().unwrap());
                                         exit(1);
                                     },
                                     Ok(path) => String::from(path.to_str().unwrap()),
-- 
2.15.0

